### ISSUE TITLE
Question: Require one of n possible commands

### ISSUE THREAD

**Issue by markusa380** [scopt/scopt#199]
I have the following code:

```scala
private val parser = new OptionParser[Execution]("sub") {
    head("sub", "0.1")

    arg[String]("name")...
      .required()
      .children(
        note(""),
        cmd("create")...,
        note(""),
        cmd("start")...,
        note(""),
        cmd("delete")...,
        note(""),
        cmd("list")...
      )
  }
```

How do I require at least on of the four possible commands?

Currently the synopsis displayed by scopt is just `sub name`, followed by a list of commands. I would like it to be `sub name create|start|delete|list [options]`.

**Comment by robertPiro** [comment]
There is no check for non-emptiness on argument lists. So providing `Seq[String]()` to the parser when the each top-level opt/cmd/arg is required or only the top-level opt/cmd/arg when each of its children is required does not throw an exception.

**Comment by shawjef3** [comment]
I'm running into this same thing. I'm wanting a mechanism where my config class has a coproduct and different commands set a different value.

**Comment by markusa380** [comment]
@shawjef3 my solution was to use the library [decline](https://github.com/bkirwi/decline) instead. It's Typelevel incubator and supports this kind of behaviour well by providing a `mapN(...)` - function.

**Comment by shawjef3** [comment]
@MarkusA380 `mapN` looks to be for products rather than coproducts.

**Comment by markusa380** [comment]
@shawjef3 you're right, my mistake, but this is also supported. Take a look at their wiki: http://ben.kirw.in/decline/usage.html
If that's not enough I have a code example for a coproduct somewhere.

**Comment by ricardogaspar2** [comment]
Yeah this is what I've been looking for. Does scopt really support it?
@markusa380 about decline... where in the usage page? Can you provide a more detailed link please?

**Comment by markusa380** [comment]
http://ben.kirw.in/decline/

**Comment by markusa380** [comment]
Decline's `Opts` have all kinds of Cats-typeclass-instances like `Product` or `Coproduct`, allowing all kinds of compositions. I don't think that's too explicitly documented.
